src/Analyzers/CSharp/Analyzers/NewLines/ConsecutiveBracePlacement/ConsecutiveBracePlacementDiagnosticAnalyzer.cs

```diff
@@ -63,6 +63,9 @@ private void Recurse(SyntaxTreeAnalysisContext context, ReportDiagnostic severit
 
                 foreach (var child in current.ChildNodesAndTokens())
                 {
+                    if (!context.ShouldAnalyzeSpan(child.FullSpan))
```

Comment:
same with this.  it feels very natural to use hve the context define (through extensions of course), where you start processing, and what parts of that subtree are relevant.  This came out nicely.